src/cryptoxide_dev_tests.rs

```diff
@@ -0,0 +1,308 @@
+
+#[cfg(test)]
+mod tests {
+    use ring::digest::{Context, SHA256};
+    use ring::{hmac, test};
+
+    use std::borrow::Borrow;
+
+    use curv::BigInt;
+
+    const SHA256_SIGNATURE_RESULT_FORMATED_STRING: &str =
+        "Signature(SHA256:3c1042fecc7664f666bd94bb1e8ea1211413dc44c4967c92f65619acfc5598c4)";
+    const EXAMPLE_STRING: &str = "hello, world";
+    const EXPECTED_HEX: &str = "09ca7e4eaa6e8ae9c7d261167129184883644d07dfba7cbfbc4c8a2e08360d5b";
+
+    /// single shot method using ring to replace
+    fn compute_digest(bytes: &[u8]) -> BigInt {
+        let mut digest = Context::new(&SHA256);
+        digest.update(&bytes);
+        BigInt::from(digest.finish().as_ref())
+    }
+
+    /// single shot method replacing ring with cryptoxide
+    fn compute_digest_replace(bytes: &[u8]) -> BigInt {
+        use cryptoxide::digest::Digest;
+        use cryptoxide::sha2::Sha256;
+        use std::iter::repeat;
+
+        // create a SHA3-256 object
+        let mut hasher = Sha256::new();
+
+        // write input message
+        hasher.input(bytes);
+
+        let mut vect_result: Vec<u8> = repeat(0u8).take(32).collect(); //TODO: need some sane way of creating &mut [u8]
```

Comment:
there are other ways, but what wrong with this one?

Reply:
umm, I expected it to be something like 
`let mut key_value = vec![0u8; 32];` 
with `vec![0; size]`, or `with_capacity` instead of iterating, copying 😄 
I think I'm doing some syntax mistake so it interprets [0u8; 32] as array of  2 elements.

Reply:
probably was due to typo [0, 32] instead of [0; 32]